lib/ruby_lsp/executor.rb

```diff
@@ -96,6 +94,7 @@ def run(request)
 
         # Run listeners for the document
         emitter = EventEmitter.new
+        folding_range = Requests::FoldingRanges.new(document.parse_result.comments, emitter, @message_queue)
```

Comment:
Not directly related to this PR, but now with 5 requests depending on this code path, I feel we'll need more sophisticated error handling so failures to propagate too much, especially if we decided to invert the response updating logic to extensions.

Reply:
Yeah, agreed. We need to think about how we could collect errors on a per listener basis.